lib/clover_deriv_quda.cu

```diff
@@ -69,42 +70,35 @@ namespace quda {
     const int& mu = arg.mu;
     const int& nu = arg.nu;
 
-    Matrix<Cmplx,3> thisForce;
-    Matrix<Cmplx,3> otherForce;
+    M thisForce, otherForce;
 
     // U[mu](x) U[nu](x+mu) U[*mu](x+nu) U[*nu](x) Oprod(x)
     {
       int d[4] = {0, 0, 0, 0};
+      M U1, U2, U3, U4, Oprod1, Oprod2;
```

Comment:
At a first glance I did not realize that `M` is the type here. Might be better to be a little more verbose in the typedef?